stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -903,6 +903,9 @@ function dot(x, y) # arbitrary iterables
 end
 
 dot(x::Number, y::Number) = conj(x) * y
+dot(::Any, ::Missing) = missing
```

Comment:
Should this be restricted to `Number`? I think we generally use `missing` to only represent missing scalar values, so `dot(missing, [1, 2])` returning `missing` might be inconsistent, because it errors if `missing` is replaced by a number.

Reply:
Makes sense. The question then is what is `dot(missing, [1, 2])` meant to throw? If we change `Any` to `Number`, it will throw a `MethodError`, where the expectation might be that we get a DimensionMismatch as with `Numbers`.

Reply:
I think a `MethodError` is reasonable here, because that's what you usually get if a function isn't implemented for `missing`. If we are worried about user experience, a more general solution might be to add a hint handler for `MethodError`s involving `Missing` explaining that missing values might have to be handled manually.